packages/mdc-ripple/_mixins.scss

```diff
@@ -169,6 +168,41 @@ $mdc-ripple-common-styles-emitted_: false !default;
   @include mdc-states-hover-opacity(map-get($opacity-map, "hover"));
   @include mdc-states-focus-opacity(map-get($opacity-map, "focus"), $has-nested-focusable-element);
   @include mdc-states-press-opacity(map-get($opacity-map, "press"));
+
+  @if $activated-color {
+    &--activated {
+      // Stylelint seems to think that '&' qualifies as a type selector here?
+      // stylelint-disable-next-line selector-max-type
+      &::before {
+        opacity: map-get($opacity-map, "activated");
+      }
+
+      @include mdc-states-base-color($activated-color);
+      @include mdc-states-hover-opacity(map-get($opacity-map, "activated") + map-get($opacity-map, "hover"));
+      @include mdc-states-focus-opacity(
+        map-get($opacity-map, "activated") + map-get($opacity-map, "focus"),
+        $has-nested-focusable-element
+      );
+      @include mdc-states-press-opacity(map-get($opacity-map, "activated") + map-get($opacity-map, "press"));
+    }
+  }
+
+  @if $selected-color {
```

Comment:
Can you create a mdc-selected-state($color, $has-nested-focusable-element) mixin which just handles selected state? It seems like its isolated logic